clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -2903,6 +2903,31 @@ static void handleSubGroupSize(Sema &S, Decl *D, const ParsedAttr &AL) {
                  IntelReqdSubGroupSizeAttr(S.Context, AL, SGSize));
 }
 
+// Handles num_simd_work_items.
+static void handleNumSimdWorkItemsAttr(Sema &S, Decl *D,
+                                       const ParsedAttr &Attr) {
+  if (D->isInvalidDecl())
+    return;
```

Comment:
Just curious, is there a reason why this check isn't performed for the surrounding function attributes, but is required here?

Reply:
I think it's better to add for other attributes as well.

Reply:
Done